### DISCUSSION TITLE
Support to nest MemoryRouter with other routers

### DISCUSSION THREAD

**Discussion by VaZark** [remix-run/react-router#9331] [Proposals]
On the website that I'm currently building, we use a form wizard within Modals. However, since I have a BrowserRouter globally, I am unable to use MemoryRouter within the app. Thus I end up needing to handle the routing and rendering for navigating by hand. 

If MemoryRouter can be nested within other routers, this micro-navigation can be completely left up to `react-router`, and form wizards would be a breeze to build for everybody.

Since MemoryRouter handles history via arrays in-memory, this shouldn't impact application functionality either and can be updated without a breaking change. 

Are there any specific reasons that the library has chosen not to allow it?

**Comment by brophdawg11** [comment]
Superseded by https://github.com/remix-run/react-router/discussions/9601
